app/uk/gov/hmrc/apiplatformoutboundsoap/controllers/OutboundController.scala

```diff
@@ -41,16 +43,29 @@ class OutboundController @Inject()(cc: ControllerComponents,
   def message(): Action[JsValue] = Action.async(parse.json) { implicit request =>
     withJsonBody[MessageRequest] { messageRequest =>
       logger.info(s"Received request to send message to CCN2. Message body is $messageRequest")
-      val codValue = messageRequest.confirmationOfDelivery match {
-        case Some(cod) => cod
-        case None =>  appConfig.confirmationOfDelivery
+      messageRequest.privateHeaders match {
+        case Some(privHeaders) =>
+          if (privHeaders.length > 5) {
+            logger.info(s"privHeaders length:${privHeaders.length}")
```

Comment:
It would be helpful to output a message stating that the message isn't being sent and why. Also, I'd possibly consider logging at WARN